src/parallel/boundaries/BoundaryHandler.cpp

```diff
@@ -0,0 +1,258 @@
+                                      std::array<double, 3> end) {
+  _globalRegionStart = start;
+  _globalRegionEnd = end;
+}
+
+void BoundaryHandler::setLocalRegion(std::array<double, 3> start,
+                                     std::array<double, 3> end) {
+  _localRegionStart = start;
+  _localRegionEnd = end;
+}
+
+void BoundaryHandler::findGlobalWallsInLocalRegion() {
+  _isGlobalWall[BoundaryUtils::DimensionType::POSX] =
+      (_localRegionEnd[0] == _globalRegionEnd[0]);
+  _isGlobalWall[BoundaryUtils::DimensionType::NEGX] =
+      (_localRegionStart[0] == _globalRegionStart[0]);
+  _isGlobalWall[BoundaryUtils::DimensionType::POSY] =
+      (_localRegionEnd[1] == _globalRegionEnd[1]);
+  _isGlobalWall[BoundaryUtils::DimensionType::NEGY] =
+      (_localRegionStart[1] == _globalRegionStart[1]);
+  _isGlobalWall[BoundaryUtils::DimensionType::POSZ] =
+      (_localRegionEnd[2] == _globalRegionEnd[2]);
+  _isGlobalWall[BoundaryUtils::DimensionType::NEGZ] =
+      (_localRegionStart[2] == _globalRegionStart[2]);
+}
+
+bool BoundaryHandler::hasInvalidBoundary() const {
+  return _boundaries.at(BoundaryUtils::DimensionType::POSX) ==
+             BoundaryUtils::BoundaryType::ERROR ||
+         _boundaries.at(BoundaryUtils::DimensionType::POSY) ==
+             BoundaryUtils::BoundaryType::ERROR ||
+         _boundaries.at(BoundaryUtils::DimensionType::POSZ) ==
+             BoundaryUtils::BoundaryType::ERROR ||
+         _boundaries.at(BoundaryUtils::DimensionType::NEGX) ==
+             BoundaryUtils::BoundaryType::ERROR ||
+         _boundaries.at(BoundaryUtils::DimensionType::NEGY) ==
+             BoundaryUtils::BoundaryType::ERROR ||
+         _boundaries.at(BoundaryUtils::DimensionType::NEGZ) ==
+             BoundaryUtils::BoundaryType::ERROR;
+}
+
+bool BoundaryHandler::hasNonPeriodicBoundary() const {
+  return _boundaries.at(BoundaryUtils::DimensionType::POSX) !=
+             BoundaryUtils::BoundaryType::PERIODIC ||
+         _boundaries.at(BoundaryUtils::DimensionType::POSY) !=
+             BoundaryUtils::BoundaryType::PERIODIC ||
+         _boundaries.at(BoundaryUtils::DimensionType::POSZ) !=
+             BoundaryUtils::BoundaryType::PERIODIC ||
+         _boundaries.at(BoundaryUtils::DimensionType::NEGX) !=
+             BoundaryUtils::BoundaryType::PERIODIC ||
+         _boundaries.at(BoundaryUtils::DimensionType::NEGY) !=
+             BoundaryUtils::BoundaryType::PERIODIC ||
+         _boundaries.at(BoundaryUtils::DimensionType::NEGZ) !=
+             BoundaryUtils::BoundaryType::PERIODIC;
+}
+
+bool BoundaryHandler::isGlobalWall(
+    BoundaryUtils::DimensionType dimension) const {
+  return _isGlobalWall.at(dimension);
+}
+
+bool BoundaryHandler::isGlobalWall(int dimension) const {
+  return isGlobalWall(BoundaryUtils::convertLS1DimsToDimensionPos(dimension));
+}
+
+void BoundaryHandler::processGlobalWallLeavingParticles(
+    ParticleContainer *moleculeContainer, double timestepLength) {
```

Comment:
Haha, with reference to the linewidth and the alignment with spaces: I was actually using `clang-format -style=llvm -i` for all my new files. I thought that if I just use something standard, it'll be easier to adopt/change it in the future. But I'll figure something out, to fix the linewidth and the alignment (if you had any other formatting related queries, I'll try adding that to that config too)

Reply:
-> #276 :)

I think in principle it's VERY desireable to use clang-format. The exact style is something we have to agree upon at some point but not here. For now try to come up with something that is as close to the rest of the code as possible:
  - Tabs instead of spaces :vomiting_face: 
  - Line width of ~120
  - return type in the same line as the function name
  - opening `{` on the same line as `if`/`for`/ect...